tests/integration/SdkTest_test.cpp

```diff
@@ -5656,11 +5656,128 @@ TEST_F(SdkTest, SyncBasicOperations)
 }
 
 
+
+struct SyncListener : MegaListener
+{
+    // make sure callbacks are consistent - added() first, nothing after deleted(), etc.
+
+    // map by tag for now, should be backupId when that is available
+
+    enum syncstate_t { nonexistent, added, enabled, disabled, deleted};
+
+    std::map<int, syncstate_t> stateMap;
+
+    syncstate_t& state(MegaSync* sync)
+    {
+        if (stateMap.find(sync->getTag()) == stateMap.end())
+        {
+            stateMap[sync->getTag()] = nonexistent;
+        }
+        return stateMap[sync->getTag()];
+    }
+
+    bool anyErrors = false;
+    void check(bool b)
+    {
+        if (!b)
+        {
+            assert(false);
```

Comment:
I think we should avoid this `assert`. Otherwise, tasks may fail prematurely rather than in the right place when the `anyErrors` is checked. Also, the `assert` will prevent the rest of tests to run. 
Remember there could be some tasks in Jenkins that run in Debug.